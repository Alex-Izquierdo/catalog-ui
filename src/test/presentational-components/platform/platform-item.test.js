import React from 'react';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store' ;
import { Provider, thunk } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications/index';
import promiseMiddleware from 'redux-promise-middleware';

describe('<PlatformItem />', () => {
  let initialProps;
  let initialState;
  const middlewares = [ thunk, promiseMiddleware(), notificationsMiddleware() ];
  let mockStore;

  const PlatformItemWrapper = ({ store, children }) => (
    <Provider store={ store }>
      <MemoryRouter>
        { children }
      </MemoryRouter>
    </Provider>
  );

  beforeEach(() => {
    initialProps = {
      id: 'Foo'
    };
    mockStore = configureStore(middlewares);
  });

  it('should render correctly', () => {
    const store = mockStore(initialState);
    const wrapper = mount(<PlatformItemWrapper  store={ store } { ...initialProps } />);
    expect(shallowToJson(shallow(wrapper))).toMatchSnapshot();
  });

  it('should call handle check callback', () => {
    const store = mockStore(initialState);
    const onToggleItemSelect = jest.fn();
    const wrapper = mount(<PlatformItemWrapper  store={ store } { ...initialProps }
      editMode onToggleItemSelect={ onToggleItemSelect } description="Foo" id="foo" />);
    wrapper.find('input').simulate('change');
    expect(onToggleItemSelect).toHaveBeenCalled();
  });
});
