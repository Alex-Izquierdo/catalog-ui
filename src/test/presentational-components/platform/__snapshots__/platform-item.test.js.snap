// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlatformItem /> should render correctly 1`] = `
<GridItem
  className=""
  lg={4}
  lgOffset={null}
  lgRowSpan={null}
  md={6}
  mdOffset={null}
  mdRowSpan={null}
  offset={null}
  rowSpan={null}
  sm={12}
  smOffset={null}
  smRowSpan={null}
  span={null}
  xl={3}
  xlOffset={null}
  xlRowSpan={null}
>
  <Card
    className=""
    component="article"
  >
    <CardHeader
      className="pcard_header"
      component="div"
    >
      <ImageWithDefault
        height="20"
        src="test-file-stub"
        width="30"
      />
    </CardHeader>
    <ServiceOfferingCardBody />
    <CardFooter
      className=""
      component="div"
    />
  </Card>
</GridItem>
`;
